docs/case_cache.md

```diff
@@ -0,0 +1,71 @@
+# Case Cache
+
+The WebGPU CTS contains many tests that check that the results of an operation
+fall within limits defined by the WebGPU and WGSL specifications. The
+computation of these allowed limits can be very expensive to calculate, however
+the values do not vary by platform or device, and can be precomputed and reused
+for multiple CTS runs.
+
+## File cache
+
+To speed up execution of the CTS, the CTS git repo holds holds pre-computed
+test cases, serialized in a set of gzip-compressed binary files under
+[`src/resources/cache`](../src/resources/cache).
+
+These files are regenerated by [`src/common/tools/gen_cache.ts`](../src/common/tools/gen_cache.ts)
+which can be run with `npx grunt run:generate-cache`.
+This tool is automatically run by the various Grunt build commands.
+
+As generating the cache is expensive (hence why we build it ahead of time!) the
+cache generation tool will only re-build the cache files it believes may be out
+of date. To determine which files it needs to rebuild, the tool calculates a
+hash of all the transitive source TypeScript files that are used to build the
+output, and compares this hash to the hash stored in
+[`src/resources/cache/hashes.json`](`../src/resources/cache/hashes.json`). Only
+those cache files with differing hashes are rebuilt.
```

Comment:
We should have a note that different versions of Node will compress differently, so people know to expect if they regenerate a file it might change unnecessarily. It makes sense to mention here, because the hash thing tries to avoid unnecessary rebuilds, but it will sometimes rebuild when not necessary.

```suggestion
those cache files with differing hashes are rebuilt.

Since source changes will sometimes change the hash without changing the generated cache,
sometimes the cache will be regenerated unnecessarily. **This is OK, but try to avoid committing
no-op regenerations - this will happen if your version of Node produces different gzip outputs
than the original committer's Node did for the same input.**
```